File: 13rd/pro_5.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = s;
    for(int i=0; i<s.size(); i++){
        if(s[i]==' ')  continue;
        int temp=s[i]+n;
        if('a'<=s[i]&&s[i]<='z'){
            if(temp>'z')   temp-=26;
        }
        if('A'<=s[i]&&s[i]<='Z'){
            if(temp>'Z')   temp-=26;
        }
        answer[i]=temp;
    }
    return answer;
}

File: 15th/pro_5.java
import java.util.*;

class Solution {
    public int[] solution(int n, long k) {
        int[] answer = new int[n];
        
        long len=1;
        for(int i=2; i<=n; i++){
            len*=i;
        }
        
        Vector<Integer> v= new Vector<Integer>();
        for(int i=0; i<=n; i++){
            v.add(i);
        }
        
        long kk=0;
        for(int i=0; i<n; i++){
            int nn=n-i;
            len/=nn;
            for(int j=1; j<=n-i; j++){
                if(k<=len*j+kk){
                    answer[i]=v.get(j);
                    v.remove(j);
                    kk+=len*(j-1);
                    break;
                }
            }
        }
        
        return answer;
    }
}